### ISSUE TITLE
Trace-Agent Fails to Start with Permission Denied Error After Upgrading to Datadog Agent `7.57.0`

### ISSUE THREAD

**Issue by LQss11** [DataDog/datadog-agent#29155]
### Description

After upgrading to Datadog Agent 7.57.0 from 7.56.2, the trace-agent fails to start due to a permission error with the UDS listener, despite having datadog.apm.socketEnabled set to false.

### Configuration

Here is the relevant portion of my `values.yaml` configuration:

```yaml
targetSystem: linux
providers:
  aks:
    enabled: true

clusterAgent:
  image:
    doNotCheckTag: true
    tag: 7.57.0
  admissionController:
    configMode: service
    enabled: true
    mutateUnlabelled: true
  env:
  - name: DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_INIT_SECURITY_CONTEXT
    value: "{\"capabilities\":{\"drop\":[\"ALL\"]},\"runAsNonRoot\":true,\"runAsUser\":10000,\"readOnlyRootFilesystem\":true,\"allowPrivilegeEscalation\":false,\"seccompProfile\":{\"type\":\"RuntimeDefault\"}}"            
datadog:
  apiKeyExistingSecret: datadog-secret
  site: datadoghq.com
  apm:
    portEnabled: true
    instrumentation:
      enabled: false
    # I was thinking that this will disable socket and use hostip if socket disabled for trace-agent
    socketEnabled: false
agents:
  containers:
    traceAgent:
      securityContext:
        runAsUser: 100
        runAsNonRoot: true
        readOnlyRootFilesystem: true
        allowPrivilegeEscalation: false

  image:
    doNotCheckTag: true
    tag: 7.57.0
```

### Error Message

The `trace-agent` logs show the following error:

```
2024-09-09 17:29:59 UTC | TRACE | CRITICAL | (pkg/trace/api/api.go:712 in func2) | Error creating UDS listener: listen unix /var/run/datadog/apm.socket: bind: permission denied
```

### Steps to Reproduce

1. Upgrade Datadog Agent to version 7.57.0.
2. Apply the above configuration.
3. Observe that the `trace-agent` fails to start with a permission denied error.

### Additional Information

- **Previous Version:** 7.56.2 (worked fine)
- **Environment:** Kubernetes on AKS
- **Relevant Environment Variables:** `DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_INIT_SECURITY_CONTEXT`
- I can see this new config parameter in the new version: [trace_agent_socket](https://github.com/DataDog/datadog-agent/blob/7.57.0/pkg/config/config_template.yaml#L2988)
### Request

Could you please help in diagnosing this issue or provide guidance on how to resolve the permission issue with the UDS listener for the `trace-agent`?

**Comment by rodehoed** [comment]
Not only on k8s, also on a RHEL 9 system here....

**Comment by FlorentClarret** [comment]
Hi @LQss11 and @rodehoed,

We prepared [a fix](https://github.com/DataDog/datadog-agent/pull/29218) and we are going to release it very soon in 7.57.1.

**Comment by LQss11** [comment]
Thank you guys for the response, I actually managed to overpass that issue using the helm chart by using this 
```yaml
agents:      
       env:
       # this works
         - name: DD_APM_RECEIVER_SOCKET
           value: "unix:///var/run/datadog/apm.socket"
        # this does not work 
        # - name: DD_APM_RECEIVER_SOCKET
        #  value: "/var/run/datadog/apm.socket"
```
I realized that by default it's going to use /var/run/datadog/apm.socket but when I specify that even though there is no /var/run/datadog/apm.socket it will still work 


after fixing that I faced more issues I think they were mostly related to this feature **Adds a new launcher to handle incoming logs from integtrations** where it uses some json file somewhere under `/opt/...` directory 

the way I managed to fix this is that am working with  different agent/cluster agent versions

**cluster-agent: 7.57.0** in order to be able to use this `DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_INIT_SECURITY_CONTEXT`
**agent: 7.56.2** to avoid having permission denied

I think the best way to not face more issues like this in the future is at least to have ci run tests using user ids instead of root (datadog unprivileged) I have already submitted another issue related to this.

#29286 [User-Specific Security Context in kubernetes Test Cases on Datadog-Agent and Cluster-Agent Pods](https://github.com/DataDog/datadog-agent/issues/29286)

**Comment by FlorentClarret** [comment]
Hello @LQss11 and @rodehoed,

We just released [Agent 7.57.1](https://github.com/DataDog/datadog-agent/releases/tag/7.57.1) with a fix for this issue.

**Comment by ichinaski** [comment]
Closing this issue given the fix is now released.
